torch/_tensor_docs.py

```diff
@@ -5392,10 +5392,13 @@ def callable(a, b) -> number
            values=tensor([[ 9,  0, 10]]),
            size=(3, 3), nnz=1, layout=torch.sparse_coo)
 
-.. method:: to_sparse(*, layout=None, blocksize=None) -> Tensor
+.. method:: to_sparse(*, layout=None, blocksize=None, dense_dim=None) -> Tensor
    :noindex:
 
-Returns a sparse tensor with the specified layout and blocksize.
+Returns a sparse tensor with the specified layout and blocksize.  If
+the :attr:`self` is strided, and the layout specified is one of
+compressed layouts, the the number of dense dimensions could be
+specified, and a hybrid sparse tensor will be created.
```

Comment:
It would make sense to support this for COO layouts as well so that
```python
tensor.to_sparse(sparse_dim=sparse_dim) == tensor.to_sparse(layout=torch.sparse_coo,
                                                            dense_dim=tensor.dim() - sparse_dim)
```
holds. From the above, we have the following implementation hint:
```python
tensor.to_sparse(layout=torch.sparse_coo, dense_dim=dense_dim) 
```
returns
```python
tensor.to_sparse(sparse_dim=tensor.dim() - dense_dim)
```